File: Tecnicatura3Java/Leccion02/BloquesInicializacion/src/domain/Persona.java
package domain;

public class Persona {
   private final int IdPersona;
    private static int contadorPersonas;
    
    static{ //Bloque de inicializacion estatico
        System.out.println("Ejecucion bloque estatico");
        ++Persona.contadorPersonas;
        //idPersona = 10; No es un atributo static por esto hay error
    }
    
    { //Bloque de inicializacion NO static (contexto dinamico)
        System.out.println("Ejecucion de bloque NO estatico");
        this.IdPersona = Persona.contadorPersonas++;    //Incrementamos el atributo
    }
    
    //Los bloques de inicializacion se ejecutan antes del constructor
    
    public Persona(){
        System.out.println("Ejecucion del constructor");
    }
    
    public int getidPersona(){
        return this.IdPersona;
    }

    @Override
    public String toString() {
        return "Persona{" + "IdPersona=" + IdPersona + '}';
    }  
}

File: Tecnicatura3JS/TSLeccion07/Persona.js
class Persona{

    static contadorPersonas = 0;

    constructor(nombre, apellido, edad){
        this._idPersona = ++Persona.contadorPersonas;
        this.nombre = nombre;
        this.apellido = apellido;
        this.edad = edad;
    }

    get _idPersona(){
        return this._idPersona;
    }

    get nombre(){
        this._nombre;
    }

    set nombre(nombre){
        this._nombre = nombre;
    }

    get apellido(){
        this._apellido;
    }

    set apellido(apellido){
        this._apellido = apellido;
    }

    get edad(){
        this._edad;
    }

    set edad(edad){
        this._edad = edad;
    }

    toString(){
        return `
        ${this._idPersona} 
        ${this._nombre} 
        ${this._apellido} 
        ${this._edad}`;
    }

    

}